### DISCUSSION TITLE
"List Index Out of Range" Error with CatPhan504 Analysis

### DISCUSSION THREAD

**Discussion by anaiska27** [jrkerns/pylinac#487] [Q&A]
Hello GitHub Community,

I hope this message finds you well. I am reaching out for assistance as I've encountered an issue while analyzing my CBCT images using Pylinac's CatPhan504 module.

Upon attempting to analyze my CBCT data, I consistently encounter the error message "list index out of range". Unfortunately, I'm unsure about the cause of this error and how to resolve it.

I suspect that the error may be related to the configuration of my CatPhan504 phantom. It's possible that my phantom setup differs from the standard configuration outlined in the Pylinac documentation.

Here's the code snippet I'm using for the analysis:
from pylinac import CatPhan504
from tkinter import filedialog

cbct_folder = filedialog.askdirectory()

mycbct = CatPhan504(cbct_folder)
mycbct.analyze()
print(mycbct.results)

I would greatly appreciate any insights or guidance on how to troubleshoot and resolve this issue. If anyone has experienced a similar problem or has expertise with Pylinac's CatPhan504 module, your help would be invaluable.

Thank you in advance for your time and assistance.

**Comment by jrkerns** [comment]
Can you attach your scan?

**Comment by anaiska27** [comment]
Yes, here it is
[18112021.zip](https://github.com/jrkerns/pylinac/files/14834591/18112021.zip)

**Reply by crcrewso** [reply]
I'm concerned about the image quality of the dataset, the first two images are adjacent slices, showing a high variance in signal to noise slice to slice. I can't even discern all of the HU regions for a 504 phantom. The third shows significant artifacts. Is it possible the dose rate used was too low?  
![image](https://github.com/jrkerns/pylinac/assets/6089951/c1022ad5-253c-4a4c-8569-d1742b528628)
![image](https://github.com/jrkerns/pylinac/assets/6089951/c68b876a-3c8a-4594-a5e9-feb5874c3809)
![image](https://github.com/jrkerns/pylinac/assets/6089951/ee7bec66-0d3a-4f7a-aa8f-751b0ef6bda5)

**Reply by anaiska27** [reply]
It is a CBCT. And I made 2 protocols : head and pelvis. I didn't try with the 2 protocols but I think I will have the same problem

**Comment by crcrewso** [comment]
I think I see the problem, there's no empty slice at the end of the phantom.

**Comment by jrkerns** [comment]
Newer versions of pylinac give a better [error message](https://github.com/jrkerns/pylinac/blob/d84ca3fb9e4f53a91189fa18e4c35384f54d8baa/pylinac/ct.py#L1820C7-L1825C14). It looks like the scan is just one slice short of being considered complete. It would be best if you could move your phantom to be more centered about the CTP404 (HU linearity) module. This can also be solved by [overriding the module locations](https://pylinac.readthedocs.io/en/latest/cbct.html#customizing-module-locations) but the former is preferred.

**Reply by anaiska27** [reply]
Thanl you a lot
Can you just explain me how to fond the module's offsets please ?

**Comment by anaiska27** [comment]
I tried to add a slice at the end but I have the same error message...
I also tried to change the module locations but same error ...

**Reply by crcrewso** [reply]
Could you explain what you mean by adding a slice at the end?

**Reply by anaiska27** [reply]
I duplicated the last slice and recreated a 3D image with

**Reply by crcrewso** [reply]
So the slice you added to the end wasn't of air but was the same image as slice 93?

**Reply by anaiska27** [reply]
Yes, I had  a 88 slices scan so I added a slice n° 89, which was a copy of the slice n°88

**Reply by crcrewso** [reply]
I can see the confusion, sorry, for the algorithm to properly detect the phantom orientation there needs to be an air slice at one end of the dataset.

**Reply by anaiska27** [reply]
Oh okay, so I need to do the same but with an air slice at the end ?

**Reply by anaiska27** [reply]
I see that in my scan, there no air slice ...

**Comment by anaiska27** [comment]
[18112021.zip](https://github.com/jrkerns/pylinac/files/14853503/18112021.zip)
Last try : I added an air slice at the beginning and at the and of the phantom but I still have the same error...

**Comment by crcrewso** [comment]
Is this phantom a Catphan504 or a Catphan500? The reason I ask is because the modules appear to be in a non-standard order for a Catphan504, and the HU module looks more like the CTP401 than the CTP404 
![image](https://github.com/jrkerns/pylinac/assets/6089951/52088516-7837-4dd8-8933-46c58fb75745)

**Reply by anaiska27** [reply]
I asked the physicists and they told me that it is a 504 but I agree with you that the module seems to be a 401...
